src/index.ts

```diff
@@ -158,89 +176,52 @@ export default class SwupA11yPlugin extends Plugin {
 
 		// Replace {variables} in template
-		const announcement = this.parseTemplate(templates.visit, { title, href, url, path });
-
-		visit.a11y.announce = announcement;
-	}
+		const announcement = parseTemplate(templates.visit, { title, href, url, path });
 
-	parseTemplate(str: string, replacements: Record<string, string>): string {
-		return Object.keys(replacements).reduce((str, key) => {
-			return str.replace(`{${key}}`, replacements[key] || '');
-		}, str || '');
+		return announcement;
 	}
 
-	handleNewPageContent() {
-		// We can't `await` nextTick() here because it would block ViewTransition callbacks
-		// Apparently, during ViewTransition updates there is no microtask queue
-		nextTick().then(async () => {
-			this.swup.hooks.call('content:announce', undefined, undefined, (visit) => {
-				this.announcePageName(visit);
-			});
-			this.swup.hooks.call('content:focus', undefined, undefined, (visit) => {
-				this.focusPageContent(visit);
-			});
-		});
-	}
-
-	announcePageName(visit: Visit) {
-		if (visit.a11y.announce) {
-			this.liveRegion.say(visit.a11y.announce);
-		}
-	}
-
-	announce = (message: string): void => {
-		this.liveRegion.say(message);
-	};
-
-	async focusPageContent(visit: Visit) {
-		// Focus disabled for this visit?
-		if (!visit.a11y.focus) return;
-
-		// Found [autofocus] element?
-		if (this.options.autofocus) {
-			const autofocusEl = this.getAutofocusElement();
-			if (autofocusEl && autofocusEl !== document.activeElement) {
-				this.swup.hooks.once('visit:end', (v) => {
-					if (v.id !== visit.id) return;
-					autofocusEl.focus();
-				});
-				return;
-			}
-		}
-
-		// Otherwise, find content container and focus it
-		const content = document.querySelector<HTMLElement>(visit.a11y.focus);
-		if (!(content instanceof HTMLElement)) return;
+	focusContentElement(selector: string) {
+		const el = document.querySelector<HTMLElement>(selector);
+		if (!(el instanceof HTMLElement)) return;
 
 		// Set and restore tabindex to allow focusing non-focusable elements
-		const tabindex = content.getAttribute('tabindex');
-		content.setAttribute('tabindex', '-1');
-		content.focus({ preventScroll: true });
+		const tabindex = el.getAttribute('tabindex');
+		el.setAttribute('tabindex', '-1');
+		el.focus({ preventScroll: true });
 		if (tabindex !== null) {
-			content.setAttribute('tabindex', tabindex);
+			el.setAttribute('tabindex', tabindex);
 		} else {
-			content.removeAttribute('tabindex');
+			el.removeAttribute('tabindex');
 		}
 	}
 
-	getAutofocusElement(): HTMLElement | undefined {
-		const focusEl = document.querySelector<HTMLElement>('body [autofocus]');
-		if (focusEl && !focusEl.closest('inert, [aria-disabled], [aria-hidden="true"]')) {
-			return focusEl;
+	focusAutofocusElement(): boolean {
+		if (!this.options.autofocus) return false;
+
+		const autofocusEl = getAutofocusElement();
+		if (autofocusEl && autofocusEl !== document.activeElement) {
+			if (autofocusEl !== document.activeElement) {
```

Comment:
This looks redundant to me. Also nitpick: I like to use `isSameNode` – but `===` is just as good.